homeassistant/components/fronius/sensor.py

```diff
@@ -179,6 +190,15 @@ class FroniusSensorEntityDescription(SensorEntityDescription):
     FroniusSensorEntityDescription(
         key="status_code",
         entity_category=EntityCategory.DIAGNOSTIC,
+        entity_registry_enabled_default=False,
+    ),
+    FroniusSensorEntityDescription(
+        key="status_message",
+        response_key="status_code",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        device_class=SensorDeviceClass.ENUM,
+        options=[opt.value for opt in InverterStatusCodeOption],
+        value_fn=get_inverter_status_message,
```

Comment:
I would suggest not to expose the complexity of different status messages. Instead I belive having a generic 'get_status_message(type)' should be preferable. It could then call the specific operations and in one place handle invalid/unknown error code responses. This should allow simplified maintenance across different types and more easy expandability for furtehr types.

Reply:
I did play around with that idea a little bit, but tbh I couldn't come up with a way that was better readable than just having 3 independent functions.

A generic factory function that has branches for those 3 types accepting a TypeVar with value restriction as argument was my best guess... 😬 
Since all of the branches need slightly different error handling (one has plain `.get()`, one needs to handle `256 <= _code <= 511` specifically and one having a default value of `InverterStatusCodeOption.INVALID`) doesn't really make it less complex / better readable.

Feel free to make a suggestion or a follow-up PR, maybe I'm just thinking in the wrong direction here.

Reply:
Yes, I thought about it as well. And honestly, there will probably be a very finite numer of cases to which it could even be extended to. Afterall, it's not like the fronius API is boundless :)